File: CS/KeepGroupsExpanded/Form1.cs
using DevExpress.Utils.Drawing;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace KeepGroupsExpanded {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e) {
            new DevExpress.XtraGrid.Design.XViewsPrinting(gridControl1);
            gridView1.Columns["Discontinued"].GroupIndex = 0;

            gridView1.ExpandAllGroups();
            gridView1.OptionsMenu.EnableGroupPanelMenu = false;
        }

        private void gridView1_CustomDrawGroupRow(object sender, DevExpress.XtraGrid.Views.Base.RowObjectCustomDrawEventArgs e) {
            DevExpress.XtraGrid.Views.Grid.ViewInfo.GridGroupRowInfo info;
            info = e.Info as DevExpress.XtraGrid.Views.Grid.ViewInfo.GridGroupRowInfo;
            info.ButtonBounds = Rectangle.Empty;
            info.GroupText = " " + info.GroupText.TrimStart();
            e.Cache.FillRectangle(e.Appearance.GetBackBrush(e.Cache), e.Bounds);
            ObjectPainter.DrawObject(e.Cache, e.Painter, e.Info);
            e.Handled = true;
        }

        private void gridView1_GroupRowCollapsing(object sender, DevExpress.XtraGrid.Views.Base.RowAllowEventArgs e) {
            e.Allow = false;
        }
    }
}

File: VB/KeepGroupsExpanded/Form1.vb
Imports DevExpress.Utils.Drawing
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms

Namespace KeepGroupsExpanded
	Partial Public Class Form1
		Inherits Form

		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			Dim TempXViewsPrinting As DevExpress.XtraGrid.Design.XViewsPrinting = New DevExpress.XtraGrid.Design.XViewsPrinting(gridControl1)
			gridView1.Columns("Discontinued").GroupIndex = 0

			gridView1.ExpandAllGroups()
			gridView1.OptionsMenu.EnableGroupPanelMenu = False
		End Sub

		Private Sub gridView1_CustomDrawGroupRow(ByVal sender As Object, ByVal e As DevExpress.XtraGrid.Views.Base.RowObjectCustomDrawEventArgs) Handles gridView1.CustomDrawGroupRow
			Dim info As DevExpress.XtraGrid.Views.Grid.ViewInfo.GridGroupRowInfo
			info = TryCast(e.Info, DevExpress.XtraGrid.Views.Grid.ViewInfo.GridGroupRowInfo)
			info.ButtonBounds = Rectangle.Empty
			info.GroupText = " " & info.GroupText.TrimStart()
			e.Cache.FillRectangle(e.Appearance.GetBackBrush(e.Cache), e.Bounds)
			ObjectPainter.DrawObject(e.Cache, e.Painter, e.Info)
			e.Handled = True
		End Sub

		Private Sub gridView1_GroupRowCollapsing(ByVal sender As Object, ByVal e As DevExpress.XtraGrid.Views.Base.RowAllowEventArgs) Handles gridView1.GroupRowCollapsing
			e.Allow = False
		End Sub
	End Class
End Namespace